parquet-column/src/main/java/org/apache/parquet/schema/Type.java

```diff
@@ -114,6 +116,11 @@ public boolean isMoreRestrictiveThan(Repetition other) {
   private final ID id;
 
   /**
+   * Pattern for matching column names that need to wrapped in backquotes.
+   */
+  private static final Pattern pattern = Pattern.compile("[^a-zA-Z\\d_-]");
```

Comment:
What requirement are you referring to?

My main concern is that if we allow special characters in column names we will allow creating Parquet files that are incompatible with most object models. Thrift and protobuf can't generate a method that contains "`" and Avro's spec disallows special characters specifically for compatibility across languages and uses (code gen, generic, etc.).

I don't think it's a good idea to add this support. The alternative is for the user to choose a column name in Hive that isn't going to break others systems, and I think that's a completely reasonable requirement.